sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_legacy/prompty/_prompty.py

```diff
@@ -368,15 +373,36 @@ async def _send_with_retries(
         should_retry: bool = True
         retry: int = 0
         delay: Optional[float] = None
+        # Make a mutable copy of the parameters
+        mutable_params = dict(params)
 
         while should_retry:
             try:
                 if delay:
                     await asyncio.sleep(delay)
 
-                response = await client.chat.completions.create(**params)
+                response = await client.chat.completions.create(**mutable_params)
                 return response
             except OpenAIError as error:
+                error_message = str(error)
+                BAD_REQUEST_ERROR_CODE = "400"
+
+                # Check for BadRequestError with unsupported parameter message
+                if (not self._is_reasoning_model and
+                        'Unsupported parameter' in error_message and
+                        BAD_REQUEST_ERROR_CODE in error_message):
+                    # Apply reasoning model parameter modifications
+                    self._logger.debug(
+                        "Detected BadRequestError with unsupported parameters. Adapting parameters for reasoning model."
+                    )
+                    # Remove unsupported parameters for reasoning models
+                    AsyncPrompty._adapt_parameters_for_reasoning_model(mutable_params)
```

Comment:
This is will result in atleast 2 rest calls whenever a reasoning model is used. This is expensive. I remeber there was some work being done on service side to make it compatible with gpt model, do we know the status of that work ?